### DISCUSSION TITLE
Pieces for Developers Pro DRAFT Pricing Outline - To be released later this year

### DISCUSSION THREAD

**Discussion by rosie-at-pieces** [pieces-app/support#104] [Announcements]
Hi everyone, we get a lot of questions on how much Pieces for Developers costs. At this point, Pieces is **completely free**, but we are planning on releasing Pieces Pro in late summer or early fall of this year. We want to be transparent as possible so please see below for our current pricing plan draft. Please keep in mind that if you are an existing user of Pieces you will get early adopter pricing! 

If you are looking for a **teams** or **enterprise plan**, please reach out directly to rosie@pieces.app or [schedule a call](https://calendar.app.google/hZUPXfWVz7assuPh6) with me. 

**Pieces Pro-Plan Features & Details:**

**🚀 Pieces Pro: $8.99/mo Empower your workflow with expanded features:**
- 🔓 Unlimited Workspaces (Free accounts limited to 2)
- 🌌 Unlimited MultiModal Materials (Free accounts limited to 250 saves)
- 💬 Unlimited Persisted Copilot Conversations (Free accounts limited to 20)
- 📧 Unlimited Copilot Messages (Free accounts limited to 200/day)
- 🏷️ Unlimited Advanced Enrichment (Tags, People, Links, Anchors, Annotations, Sensitives, etc.)
- 🔄 Unlimited Advanced Material Transforms
- Convert PNG to SVG, generate code from screenshots/videos, and more. Exciting new support for Front-End Developer materials is on the way!
- 🔗 Granular Link Sharing Permissions (Email-based sharing)
- ☁ Second Generation Virtual Private Clouds (Reduced cold start times)
- 📊 Unlimited Workflow Activity Events (Free accounts have ~45 days of history or 5000 events)
- 🔙 Unlimited Git Based Version Control + Rollbacks for your Pieces instance
- 🛠️ 24/7 Support & Live Troubleshooting with the Core-Engineering Team

From there, we're introducing an "add-on" system for specialized feature sets, allowing us to keep the Pro Plan affordable for everyone.

**☁ Cloud Add-On: $5.99/mo Enhance your storage and access capabilities:**
- 🔐 Unlimited Backups (Copilot Conversations, Feed, Saved Materials, Workflow Context)
- ⚡ Always-on Virtual Private Clouds (Fastest data transfer rates & link sharing)
- 📂 Advanced VPC Mounted Storage (Uses Near-Shore FileStore for better reliability)
- 🔄 Unlimited Real-Time Cross-Device Syncing
- 🌐 Personalized Cloud Domains (e.g., tsavo.pieces.cloud)
- 📱 Mobile Access to your workflow, available on iOS, Android, and Web.
- 👑 Priority Support & Live Troubleshooting with Founders, PMs, and QA Team

**🤖 Advanced Copilot Add-On: $11.99/mo Unlock next-level AI assistance:**
- 🧠 Unlimited Usage of Advanced LLMs (OpenAI GPT-4, Google Palm 2, Google Gemini, Anthropic Claude)
- 🔮 Unlimited Persisted/Triangulated Workflow Context (Seamless switching between environments)

**🛡️ Advanced Security & Privacy Add-on: $8.99/mo Fortify your digital environment:**
- 🔒 Unlimited Usage of Air-Gapped Enrichment & LLMs (Mistral, CodeLlama, Falcon, etc.)
- 🌉 Bring Your Own Cloud LLM Provider (Integrate existing enterprise-level services)
- 📡 P2P LAN/VPC Sharing (Secure local network sharing)
- 🔑 Military-Grade Encryption with advanced decryption options
- 🏠 Self-Hosted Cloud Server Instances

**🔧 Extendability + Internal Tooling Add-On: Pay as You Go Unlock the full power of Pieces with direct access to our suite of tools and services:**
- 🔨 Unlimited Local Access to Pieces OS Endpoints
- Dive into a world of possibilities with direct access to functionalities like localhost:1000/tagify, localhost:1000/ocr, localhost:1000/qGPT, and more for just $0.00008 per request.
- 🌐 Proxied+Protected External Access
- Interact with your Pieces OS instances securely with operations like GET tsavo.pieces.cloud/assets, tsavo.pieces.cloud/conversations/create, at $0.0002 per request.
- 💼 Priority Support and Integration Assistance
- Get direct assistance from the creators: our Founders and Core-Engineering Teams are here for you.

**Comment by dmwyatt** [comment]
Do I understand correctly that you're planning on charging an extra fee to use on-device models like Mistral?

**Reply by rosie-at-pieces** [reply]
Hi @dmwyatt 👋 not yet, and not definitely. This is a draft pricing plan that we put out to get feedback from users like you. Would love to know your thoughts! 

Pieces is 100% free right now including all LLMs. We do have to think about long- term sustainability however so this is one way we are  considering charging. We want Pieces to always have a free tier, and charging for advanced models may be part of that.

**Reply by dmwyatt** [reply]
Right, I got that you're not yet...I'm currently using it!

I just think that users will balk at the idea of having to pay an on-going fee for local models running on their own device. I get that there's maintenance costs to all software, but that's just going to be the perception.

**Reply by rosie-at-pieces** [reply]
That makes complete sense! We’ll take it back to the drawing board.

Thank you for the input and for being a Pieces user!!

**Reply by nigelp** [reply]
You could try a one time flat fee and see how that flies as an idea?

**Reply by UselessParadox** [reply]
Yes a one time flat fee and might a yearly one ... as long the user running their own LLMs

**Comment by Grenish** [comment]
It would be somewhat disappointing to discover that PiecesAI is no longer accessible for free. Many users rely on it for their daily tasks, as it is the sole AI tool that offers top features for free, unlike other paid alternatives.
As a developer, I myself use it daily and find it really good and comfortable. The support is excellent, and we have access to all the latest models available, whether in the cloud or on-device.

**Reply by rosie-at-pieces** [reply]
Hey @Grenish - thank you, this is great feedback. I love hearing that you use it every day (and if you ever want to chat and do a 1:1 user interview just email me rosie@pieces.app)!! 

We plan to always have a free tier, so users can continue to rely on it for their every day tasks even if they can't pay for the Pro plan. I do understand that it would be disappointing to no longer have access to all the models; unfortunately, that is likely not sustainable for us as a company long-term. But I would be interested to know, if you have to carve out some models that stay free, which ones would they be?

And don't worry, we will absolutely continue with the top tier support 🤝

**Reply by UselessParadox** [reply]
I mean computing cost is something what needs to be paid as well maintainer fee for the software itself. If you can track somehow where someone if they made a successful project they have to pay a monthly fee .. something like unreal offer ..

**Comment by rawwerks** [comment]
@rosie-at-pieces - my two cents on pricing options: 

i think that having an "at cost" (bring-your-own-key) option goes a long way to building trust. cursor.sh has done this with great success. i don't think adding margin to inference is a great business model and it just makes people want to find a way around the product. or makes people start to do stupid games to figure out the cheapest way to get the lowest inference costs. 

what i like about cursor's business model, which again has been very successful building trust with developers, is that they have advanced features that are only available to subscribers. so people who are cost-conscious or just generally skeptical of SAAS can use the basic product with their own key at cost, but the best features and advanced features are only for paying customers. 

from my personal experience with cursor, it was this "generosity" of the bring-your-own-key combined with the fact that their product is truly awesome, that convinced me to go from being a free user (with my own key) to being an enterprise customer paying the highest rate for me and my whole team.  

related to the overall business model but not specifically pricing: because you are storing people's notes, i would also say "don't be evernote". show people there is a way to get their data out.

**Reply by Grenish** [reply]
While having advanced features for subscribers is an excellent strategy, it’s essential to ensure that the basic offering still delivers substantial value. Users might be more inclined to upgrade if they feel the free version is already highly beneficial. That's what I think.

**Comment by Idrinth** [comment]
Looks like I would upgrade to 8.99$ x2 to have local only LLMs. Sounds relatively reasonable to me, especially the ability to share locally is a huge factor when you use pieces on 3 devices.

**Comment by hra42** [comment]
I wanted to sum up some discussion on Discord from the last 2/3 weeks.

1. multimodal saves limited to 250:
    - What exactly is a MultiModal Material (saves)? I would guess a snippet that's based on a picture?
2. Bring your own API Keys:
    - Will that be covered by the Advanced Security & Privacy Add-on?
3. Free model:
    - How is the free plan supposed to cover costs? Some users may fear that with such a great free plan, pieces might want to sell their data.
4. Timeline:
    - Do you have a timeline for the introduction of paid plans? (H2 2024 or later?)

**Comment by iamscottb** [comment]
The original discord post about pricing also mentioned bundling of add-ons.  The bundling would include compounded discounts depending on the number of add-ons bundled.  I do not see it mentioned above. 
Is this still in the plans?

**Comment by hesswebtech** [comment]
The part of the draft mentioning advanced LLM access/switching seems odd from a user perspective to me. I totally get it from a company cost side of things, but two things jumped out to me...

1) It may be only me, but I am used to being able to switch LLMs through other bot-building services I have used. So I have come to see it as a part of the basic tier.

2) The usage of the term "advanced". This may simply come down to marketing optics, but I want to access the advanced LLMs out of the gate. I have to ask myself, "Why would I want only basic?" I understand the LLM differences, but entry level users may not get that. Again, it may be a marketing/educational issue.

Possible solution...have LLM switching be available to everyone out of the gate. Free LLMs only available to free users of pieces. Once someone pays, they then get access to paid LLMs. You would probably have to work out how to make it plain to the dev how much they can access (number of messages, number of 'X' they can do a month) and give options for expansion of that usage. There is complexity due to the various pricing models of the LLMs.

Anyway...just my two cents... ;)

**Comment by Gokilp** [comment]
Amazing 🤩

**Comment by CPX10** [comment]
Can I add, some **lifetime payment** cause pay monthly an app, has issues with the banks and some apps they don´t renfund you, and not an expensive lifetime payment

**Reply by hra42** [reply]
I doubt a lifetime membership would be possible, LLMs are expensive and you have high risk of running into cost issues, no matter how high the price is.

I'd rather say that there should be a cheap plan <$10 that have some form of low usage tier.

**Comment by domedevep** [comment]
try to add a bundle price of all of those extra capabilities, and please, don't just use stripe, in some countries is hard to pay with it, accept Paypal.

**Reply by AmeyKuradeAK** [reply]
Then they have options to add different gateways. 
1. India - Razorpay
2. other - stripe
3. Paypal

**Reply by mark-at-pieces** [reply]
@domedevep We are currently in the initial phases of adding in our payment pipeline, we will accept Paypal 100% and are looking at alternative tools other than Stripe for instance, https://www.paddle.com/

**Reply by hra42** [reply]
@mark-at-pieces are you able to share an update on pricing?

**Comment by padparadscho** [comment]
Here's my take:

Consider offering a bundled option, like Pieces Pro with add-ons at a discounted price. Additionally (just a personal preference), it would be great to have a yearly plan available.

**Comment by iamscottb** [comment]
There also needs to be some pricing around team/enterprise plans if an organization wishes to have multiple users.

**Reply by hra42** [reply]
@tsavo-at-pieces you might wanna have a chat about this?

**Comment by BrandenStoberReal** [comment]
Many other users including myself would appreciate a grandfathered free plan due to being early users/beta testers. It doesn't have to be anything major, but it helps give back to the community that got pieces rolling.

**Reply by hra42** [reply]
afaik: the pro plan will be free for the first year for all early access users!

**Comment by trrahul** [comment]
Consider adding Purchase Power Parity-based adjustments to the pricing, which could attract more paid users like me. e.g. [Codecrafters.io](https://app.codecrafters.io/pay) has the same.
